flink-examples/flink-java-examples/src/main/java/org/apache/flink/examples/java/misc/testing/CollectTesting.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.examples.java.misc.testing;
+
+import org.apache.flink.api.java.DataSet;
+import org.apache.flink.api.java.ExecutionEnvironment;
+
+public class CollectTesting implements java.io.Serializable {
```

Comment:
Few comments on the test:
 
 - Tests should not print messages to sysout.

 - Tests that include actor messaging need to be executed with multiple actor systems. Otherwise, no message serialization happens. You can start a `ForableFlinkMiniCluster` and tell it to use different actor systems for JobManager and TaskManager. The simplest thing would be to simply add the "oversized accumulator" test to https://github.com/apache/flink/blob/master/flink-tests/src/test/java/org/apache/flink/test/misc/MiscellaneousIssuesITCase.java

 - Tests need to be deterministic on multiple setups. That includes hard-setting the degree of parallelism. If not explicitly specified, the parallelism becomes the number of CPU cores in the local executor, making the test behave differently on different testing machines.